src/filters.rs

```diff
@@ -383,4 +404,9 @@ mod tests {
         assert_eq!(unit!(last, Array(vec![tos!("test"), tos!("last")])), tos!("last"));
         assert_eq!(unit!(last, Array(vec![])), tos!(""));
     }
+
+    #[test]
+    fn unit_date() {
+        assert_eq!(unit!(date, tos!("13 Jun 2016 02:30:00 +0300"), &[tos!("%Y-%m-%d")]), tos!("2016-06-13"));
```

Comment:
Could you add tests that pass in wrong values for the three error conditions in your function? There are some (unfortunately not enough) examples in the tests above.